client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -199,16 +199,21 @@ public boolean replaceWithVersion(K key, V newValue, long version, long lifespan
       }
    }
 
+   @Override
+   public CompletableFuture<MetadataValue<V>> getWithMetadataAsync(K key) {
+      return getWithMetadataAsync0(key, dataFormat);
+   }
+
    @Override
    public MetadataValue<V> getWithMetadata(K key) {
-      return getWithMetadata(key, dataFormat);
+      return await(getWithMetadataAsync(key));
    }
 
-   private MetadataValue<V> getWithMetadata(K key, DataFormat dataFormat) {
+   protected CompletableFuture<MetadataValue<V>> getWithMetadataAsync0(K key, DataFormat dataFormat) {
```

Comment:
Other methods don't have the `0` suffix, why here?

Reply:
oops, a leftover